core/src/main/scala/kafka/server/DelayedRemoteListOffsets.scala

```diff
@@ -31,7 +31,7 @@ import java.util.concurrent.TimeUnit
 import scala.collection.{Map, mutable}
 import scala.jdk.CollectionConverters._
 
-case class ListOffsetsPartitionStatus(var responseOpt: Option[ListOffsetsPartitionResponse] = None,
+case class ListOffsetsPartitionStatus(@volatile var responseOpt: Option[ListOffsetsPartitionResponse],
```

Comment:
Intellij shows "Class parameter shadows superclass var". How to fix this warning?

<img width="1184" alt="Screenshot 2024-10-13 at 1 14 39 AM" src="https://github.com/user-attachments/assets/1c1618fd-8a2d-4732-b2dc-cdffb7b5e511">

Reply:
Probably this conflicts with the case class. Maybe use a non-case class?

Reply:
Used a non-case class with object to overcome the warning.